docs/doc.md

```diff
@@ -0,0 +1,186 @@
+# Application
+
+This doc explains high-level app design with a structural and functional view. 
+
+## Structural View
+
+```mermaid
+---
+title: Layer View
+---
+graph
+    UI --> Data
+    UI --> Controllers --> Data
+    x[Cross-Layer]
+```
+
+`UI` implements everything related to interaction with the user: screens, animations, navigation between the screens.
+
+`Controllers` implement processes and business logic functionality of the app, uses `Data` layer's objects.
+
+`Data` implements access to the persistence, secure storage, as well as API models for various remote services. 
```

Comment:
One of the categories describes in the graph and down below is called "Cross-Layer" but it's missing from this brief description. Probably we should add it here too.